crates/miden-lib/src/account/faucets/mod.rs

```diff
@@ -220,6 +223,24 @@ impl TryFrom<&Account> for BasicFungibleFaucet {
 // FUNGIBLE FAUCET
 // ================================================================================================
 
+pub struct FungibleFaucet;
+
+impl FungibleFaucet {
+    const ISSUANCE_ELEMENT_INDEX: usize = 3;
+
+    pub fn get_issuance_from_slot(slot: &Word) -> Felt {
+        slot[Self::ISSUANCE_ELEMENT_INDEX]
+    }
+
+    pub fn get_issuance_from_account(account: &Account) -> Result<Felt, FungibleFaucetError> {
+        let slot = account
+            .storage()
+            .get_item(FAUCET_STORAGE_DATA_SLOT)
+            .map_err(|_| FungibleFaucetError::InvalidStorageOffset(FAUCET_STORAGE_DATA_SLOT))?;
+        Ok(Self::get_issuance_from_slot(&slot))
+    }
```

Comment:
We should check if the `account` is in fact a fungible faucet account and return an error if not.